spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -149,6 +149,79 @@ describe("Router", function()
     end)
   end)
 
+  describe("URI arguments (querystring)", function()
+
+    setup(function()
+      insert_apis {
+        {
+          name = "api-1",
+          upstream_url = "http://httpbin.org",
+          hosts = { "example.com" },
+        },
+        {
+          name = "api-2",
+          upstream_url = "http://localhost:9999",
+          hosts = { "localhost" },
+        },
+      }
+
+      assert(helpers.start_kong {
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      })
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("preserves URI arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get",
+        query  = {
+          foo   = "bar",
+          hello = "world",
+        },
+        headers = {
+          ["Host"] = "example.com",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.equal("bar", json.args.foo)
+      assert.equal("world", json.args.hello)
+    end)
+
+    it("does proxy an empty querystring if URI does not contain arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get?",
+        headers = {
+          ["Host"] = "localhost",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.matches("/get%?$", json.vars.request_uri)
```

Comment:
We test this behavior with our custom template since both httpbin and mockbin strip the querystring delimiter if it is empty.